SPEC.md

```diff
@@ -182,6 +183,33 @@ Contents give more information about the issue's check, including a description
 
 Some engines require the ability to refer to other source locations. For this reason, the Issue type has an optional `other_locations` field, which is an array of other `Location` items that this issue needs to refer to.
 
+## Versioning
+
+This specification is versioned. The current version is [in the repository](https://github.com/codeclimate/spec/blob/master/VERSION). Engines declare the version of the specification they are compatible with in the manifest file, described below.
+
+## Manifest File
+
+All engines must include a `manifest.json` file at `/manifest.json`. This file, known as the manifest, includes information that is necessary for the analysis runtime. Here is an example of a manifest:
+
+```
+{
+  "name": "govet",
+  "maintainer": {
+    "name": "Michael R. Bernstein",
+    "email": "mrb@codeclimate.com",
+  },
+  "spec_version": "0.0.1",
+}
+```
+
+The fields in the manifest include:
+
+* `name` (`String`) - the name of the package.
+* `maintainer` (`Object`)
+  * `name` (`String`) - the name of the maintainer
+  * `email` (`String`) - the email address of the maintainer
+* `spec_version` (`String`) - the most recent version of the specification which this engine supports
```

Comment:
We can actually play this out using the `include_path` feature as an example:

- Pretend it's a few weeks ago, before we built the `include_paths` feature
- Pretend at that time the spec was v0.1
- Pretend we had this engines.json thing and all engines have `spec_version: 0.1`
- We implement `include_paths` and deprecate `exclude_paths` in SPEC
- What happens?
- We actually want to stop supporting `exclude_paths` in CLI/Builder
- What happens?